relay-general/src/pii/convert.rs

```diff
@@ -39,7 +39,12 @@ pub fn to_pii_config(
     }
 
     if datascrubbing_config.scrub_ip_addresses {
+        // legacy(?) scrubs all fields that are known to have IPs regardless of actual content
         applications.insert(KNOWN_IP_FIELDS.clone(), vec!["@anything:remove".to_owned()]);
+
+        // checks actual contents of all fields and scrubs where there is an IP address
+        let wildcard = SelectorSpec::Path(vec![SelectorPathItem::DeepWildcard]);
```

Comment:
Instead of pushing to `applications` with a custom selector spec for IP scrubbing, I would simply push `@ip:replace` to `applied_rules` (see code further below). It will then use the same selector as the other default rules, which I think is what we want.